xpsi/Everywhere.py

```diff
@@ -155,6 +181,32 @@ def __init__(self,
         self.time_invariant = time_invariant
         self._integrator_toggle = _integrator_toggle
 
+    @property
+    def atm_ext(self):
+        """ ... """
+        return self._atm_ext
+
+    @atm_ext.setter
+    def atm_ext(self,extension):
+        if extension=="BB":
+            self._atm_ext = 1
+        elif extension=="Num4D":
+            self._atm_ext = 2
+        elif extension=="user":
+            self._atm_ext  = 3
+        else:
+            raise TypeError('Got an unrecognised atm_ext argument. Note that the only allowed '
+                            'atmosphere options are at the moment "BB", "Num4D", and "user".')
+
+    @property
+    def beam_opt(self):
+        """ ... """
+        return self._beam_opt
+
+    @beam_opt.setter
+    def beam_opt(self,option):
+        self._beam_opt = option
```

Comment:
There should also be a`TypeError` if `beam_opt` is different from the option 0, 1, 2, 3, no ?

Reply:
Good point! I can add that.

Reply:
fixed with 7cfa468864ddf67b0c02d096ce774076b26c7607